imagenet/main.py

```diff
@@ -83,12 +94,49 @@ def main():
         warnings.warn('You have chosen a specific GPU. This will completely '
                       'disable data parallelism.')
 
-    args.distributed = args.world_size > 1
+    args.distributed = args.world_size > 1 or args.multiprocessing_distributed
+
+    ngpus_per_node = torch.cuda.device_count()
+    if args.multiprocessing_distributed:
+        # python version check
+        if not ((sys.version_info[0] == 3 and sys.version_info[1] >= 4) or
+                sys.version_info[0] > 3):
+            raise RuntimeError("Requires python 3.4 or higher to use "
```

Comment:
can you move this into `mp.spawn`. There's a large error message here, a lot of logic, all of it not really driving a short example.

`mp.spawn` should be raising such an error anyways, and suggest to people the error that you wrote here.

Reply:
addressed